encord/objects/space.py

```diff
@@ -0,0 +1,134 @@
+from abc import ABC
+from enum import StrEnum
+from typing import Iterable, Optional, TypeVar
+
+from encord.objects import LabelRowV2, Object, ObjectInstance, Shape
+from encord.objects.coordinates import AudioCoordinates, Coordinates, TextCoordinates
+from encord.objects.frames import Frames, Range
+
+
+class SpaceType(StrEnum):
+    """Fixed set of available space types"""
+
+    AUDIO = "AUDIO"
+    TEXT = "TEXT"
+    VISION = "VISION"
+    POINT_CLOUD = "POINT_CLOUD"
+
+
+SpaceT = TypeVar("SpaceT", bound="Space")
+
+
+class Space(ABC):
+    def __init__(
+        self, id: str, title: str, space_type: SpaceType, parent: LabelRowV2, layout_key: Optional[str] = None
+    ):
+        self.id = id
+        # Title would probably be:
+        # name of storage_item, OR
+        # key in data group
+        self.title = title
+        self.space_type = space_type
+        self.parent = parent
+
+        # This will only be not None when space is a data group child
+        # This refers to the key of the data group child in the layout (e.g. video-left)
+        self.layout_key = layout_key
```

Comment:
This looks suspect on 2 counts.
First - you probably want to set it as a layout directly, not just an id.
Second - this seems like something that does not necessarily belong to the space? We should be able to figure out if it's a group by it's parent.